test/testUtils/MessageCheckViewSpec.scala

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package testUtils
+
+import org.jsoup.Jsoup
+import org.jsoup.nodes.Document
+import play.api.i18n.{Lang, MessagesApi}
+import play.twirl.api.Html
+import testOnly.views.html.MessageCheckView
+
+class MessageCheckViewSpec extends TestSupport {
+
+  val messageCheckView: MessageCheckView = app.injector.instanceOf[MessageCheckView]
+  lazy val msgs: MessagesApi = app.injector.instanceOf[MessagesApi]
+  implicit val lang: Lang = Lang("cy")
+
+  class Setup(language: String) {
```

Comment:
* trait is more preferable way to avoid code duplication;
* under Scala 2.x you can define field in the trait if you like to pass a parameter etc;